src/main.rs

```diff
@@ -1202,115 +1203,113 @@ fn cleanup_autogenerated_file(path: &Path) {
-//         if cmd_base.startswith(prefix):
-//             return
-//
-//     # Ignore the millions of links to BUILTIN(1)
-//     if 'BUILTIN 1' in manpage or 'builtin.1' in manpage:
-//         return
-//
-//     # Clear the output list
-//     built_command_output[:] = []
-//
-//     if DEROFF_ONLY:
-//         parsers = [TypeDeroffManParser()]
-//     else:
-//         parsers = [Type1ManParser(), Type2ManParser(), Type4ManParser(), Type3ManParser(), TypeDarwinManParser(), TypeDeroffManParser()]
-//     parsersToTry = [p for p in parsers if p.is_my_type(manpage)]
-//
-//     success = False
-//     if not parsersToTry:
-//         add_diagnostic(manpage_path + ": Not supported")
-//     else:
-//         for parser in parsersToTry:
-//             add_diagnostic('Trying %s' % parser.__class__.__name__)
-//             diagnostic_indent += 1
-//             success = parser.parse_man_page(manpage)
-//             diagnostic_indent -= 1
-//             # Make sure empty files aren't reported as success
-//             if not built_command_output:
-//                 success = False
-//             if success:
-//                 PARSER_INFO.setdefault(parser.__class__.__name__, []).append(CMDNAME)
-//                 break
-//
-//         if success:
-//             if WRITE_TO_STDOUT:
-//                 output_file = sys.stdout
-//             else:
-//                 fullpath = os.path.join(output_directory, CMDNAME + '.fish')
-//                 try:
-//                     output_file = codecs.open(fullpath, "w", encoding="utf-8")
-//                 except IOError as err:
-//                     add_diagnostic("Unable to open file '%s': error(%d): %s" % (fullpath, err.errno, err.strerror))
-//                     return False
-//
-//             built_command_output.insert(0, "# " + CMDNAME)
-//
-//             # Output the magic word Autogenerated so we can tell if we can overwrite this
-//             built_command_output.insert(1, "# Autogenerated from man page " + manpage_path)
-//             # built_command_output.insert(2, "# using " + parser.__class__.__name__) # XXX MISATTRIBUTES THE CULPABILE PARSER! Was really using Type2 but reporting TypeDeroffManParser
-//
-//             for line in built_command_output:
-//                 output_file.write(line)
-//                 output_file.write('\n')
-//             output_file.write('\n')
-//             add_diagnostic(manpage_path + ' parsed successfully')
-//             if output_file != sys.stdout:
-//                 output_file.close()
-//         else:
-//             parser_names =  ', '.join(p.__class__.__name__ for p in parsersToTry)
-//             #add_diagnostic('%s contains no options or is unparsable' % manpage_path, BRIEF_VERBOSE)
-//             add_diagnostic('%s contains no options or is unparsable (tried parser %s)' % (manpage_path, parser_names), BRIEF_VERBOSE)
-//
-//     return success
+fn parse_manpage_at_path(
+    manpage_path: &Path,
```

Comment:
```rust
fn parse_manpage_at_path<P: AsRef<Path>(
    manpage_path: P,
    output_directory: P,
    ..
}
```
?

Reply:
I don't get what you want to convey but I guess that you ask why is there additional fields not in the python version.

It is because some of the variables are available globally, so we need to make it locally by passing it through function.

Reply:
i mean `&Path` is a more restrictive type than `AsRef<Path>` because `str`, `String`, `Path`, and `PathBuf` implement it.

Reply:
Yes, it is more restrictive. But we don't need to use generics here, we are not exposing a library, until we need it we could keep it as `&Path`. This could be the reason compilation time blow up, I think taking what we need to enough.

Reply:
Fair